controllers/openstackbaremetalset_controller.go

```diff
@@ -523,6 +531,30 @@ func (r *OpenStackBaremetalSetReconciler) provisionServerCreateOrUpdate(
 	return provisionServer, nil
 }
 
+func (r *OpenStackBaremetalSetReconciler) provisionServerDelete(
+	ctx context.Context,
+	instance *baremetalv1.OpenStackBaremetalSet,
+) error {
+	l := log.FromContext(ctx)
+
+	provisionServer := &baremetalv1.OpenStackProvisionServer{
```

Comment:
Should we also compare `ownerRef` of the `instance.Spec.ProvisionServerName` ProvisionServer  before deleting? Though normally won't be the case, it may be remotely possible that user sets the same name as created earlier for the baremetalset and we will end up deleting it and fail?

Reply:
To make sure I understand: you mean if the user goes into the `OpenStackBareMetalSet` and changes `provisionServerName` to be `<instance name>-provisionserver`?

Reply:
yeah, to an existing ProvisionServer, which could be the one created for this OpenStackBareMetalSet. Do you think that won't cause any issue.

Reply:
I think that's a good point.  If an `OpenStackProvisionServer` was auto-generated at first (because `provisionServerName` was empty-string), and then the user for some reason set `provisionServerName` to that `OpenStackProvisionServer`, then we would have a problem given this new logic.  It would delete the `OpenStackProvisionServer` and then try to use it, which would just cause everything to hang until the user manually recreated an `OpenStackProvisionServer` with `<OSBMS name>-provisionserver` as its name.

We should protect against that somehow.

Reply:
Okay, I added a check to avoid deleting the `OpenStackProvisionServer` in these cases:

1. User updates `OpenStackBareMetalSet` and sets `provisionServerName` to the previous auto-generated OpenStackProvisionServer
2. User pre-created an `OpenStackProvisionServer` beforehand with a name that matches the one that would be auto-generated by this controller